### PR TITLE
Toggle creation of CloudWatch Logs components

### PR DESCRIPTION
**Pull request by chrischildresssg**
## Change description

> This update makes the creation of (and integration with) CloudWatch Logs components optional.

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

> Fix [SGINT-875](https://stratusgrid.atlassian.net/browse/SGINT-875) 

### Development

- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [x] "Ready for review" label attached and reviewers assigned
- [x] Changes have been reviewed by at least one other contributor
- [x] Pull request linked to task tracker where applicable

[SGINT-875]: https://stratusgrid.atlassian.net/browse/SGINT-875?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: cloudwatch.tf [hunk 1]
```diff
@@ -31,14 +33,15 @@ data "aws_iam_policy_document" "cloudwatch_logs_role" {
       "logs:PutLogEvents",
     ]
     resources = [
-      "${aws_cloudwatch_log_group.cloudtrail.arn}:*",
+      var.enable_cloudwatch_logs_for_cloudtrail ? "${aws_cloudwatch_log_group.cloudtrail[0].arn}:*" : ""
```
Comment (wesleykirklandsg) [inline cloudwatch.tf]: Shouldn't this entire policy be placed in a count?

File: cloudwatch.tf [hunk 2]
```diff
@@ -31,14 +33,15 @@ data "aws_iam_policy_document" "cloudwatch_logs_role" {
...
     ]
     sid = "AWSCloudTrailLogging"
   }
 }
 
 resource "aws_iam_role_policy" "cloudwatch_logs" {
+  count  = var.enable_cloudwatch_logs_for_cloudtrail ? 1 : 0
   name   = "cloudwatch-logs"
   policy = data.aws_iam_policy_document.cloudwatch_logs_role.json
```
Comment (wesleykirklandsg) [inline cloudwatch.tf]: If the above becomes a count, use a ternary here

File: inputs.tf [hunk 3]
```diff
@@ -9,10 +9,10 @@ variable "name_suffix" {
   default     = ""
 }
 
-variable "input_tags" {
-  description = "Map of tags to apply to resources"
-  type        = map(string)
-  default     = {}
+variable "enable_cloudwatch_logs_for_cloudtrail" {
+  default     = false
```
Comment (wesleykirklandsg) [inline inputs.tf]: If we change the module major version this is fine, otherwise it should be `true`

File: cloudwatch.tf (hunk 1 above)
Comment (dickc-sg) [inline cloudwatch.tf]: It doesn't appear to me that `""` is a valid value for `resources`. Must be a `*`, a string, or a set of strings: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_grammar.html#policies-grammar-bnf

Also, is there a reason you placed the conditional logic inside `resource`? If you haven't already, I would consider moving the logic to the data source level and use a `count` argument:
```hcl
data "aws_iam_policy_document" "cloudwatch_logs_role" {
  count = var.enable_cloudwatch_logs_for_cloudtrail == true ? 1 : 0

  statement {
    actions = [
      "logs:CreateLogStream",
      "logs:PutLogEvents",
    ]
    resources = [
      "${aws_cloudwatch_log_group.cloudtrail[0].arn}:*"
    ]
    sid = "AWSCloudTrailLogging"
  }
}
```
and then
```hcl
resource "aws_iam_role_policy" "cloudwatch_logs" {
  count  = var.enable_cloudwatch_logs_for_cloudtrail ? 1 : 0
  // ...
  policy = data.aws_iam_policy_document.cloudwatch_logs_role[0].json
  // ...
}
```

https://developer.hashicorp.com/terraform/language/data-sources#multiple-resource-instances

**Review by dickc-sg — CHANGES_REQUESTED** [review]

**Review by dickc-sg — DISMISSED** [review]
LGTM. I think this warrants a new major release version, as the default behavior has changed.

**Review by dickc-sg — APPROVED** [review]

**Review by wesleykirklandsg — APPROVED** [review]
LGTM

**Pull request merged**